test/irmin-pack/test_pack.ml

```diff
@@ -280,6 +284,65 @@ module Pack = struct
         Alcotest.fail "Add after closing the index should not be allowed")
       (function I.Closed -> Lwt.return_unit | exn -> Lwt.fail exn)
 
+  (** Index can be flushed to disk independently of pack, we simulate this in
+      the tests using [Index.filter] and [Index.flush]. *)
```

Comment:
Can you add some explanation about why this might be an issue? E.g. "Regression test for PR 1008 in which values were indexed before being reachable in Pack."

Reply:
Added your suggested comment.